lib/xray_creator.py

```diff
@@ -107,78 +116,76 @@ def get_results_create(self):
                         completed_xray_formats = [fmt for fmt, info in book.xray_formats_not_failing()]
                         fmts_completed.append('X-Ray ({0})'.format(', '.join(completed_xray_formats)))
             if self._create_send_author_profile:
-                if book.author_profile_status == book.FAIL:
+                if book.author_profile_status == FAIL:
                     fmts_failed.append('Author Profile: {0}'.format(book.author_profile_status_message))
                 else:
                     fmts_completed.append('Author Profile')
             if self._create_send_start_actions:
-                if book.start_actions_status == book.FAIL:
+                if book.start_actions_status == FAIL:
                     fmts_failed.append('Start Actions: {0}'.format(book.start_actions_status_message))
                 else:
                     fmts_completed.append('Start Actions')
             if self._create_send_end_actions:
-                if book.end_actions_status == book.FAIL:
+                if book.end_actions_status == FAIL:
                     fmts_failed.append('End Actions: {0}'.format(book.end_actions_status_message))
                 else:
                     fmts_completed.append('End Actions')
 
             if len(fmts_completed) > 0:
-                self._create_completed.append('{0}: {1}'.format(book.title_and_author, ', '.join(fmts_completed)))
+                create_completed.append('{0}: {1}'.format(book.title_and_author, ', '.join(fmts_completed)))
             if len(fmts_failed) > 0:
-                self._create_failed.append('{0}:'.format(book.title_and_author))
+                create_failed.append('{0}:'.format(book.title_and_author))
                 for fmt_info in fmts_failed:
-                    self._create_failed.append('    {0}'.format(fmt_info))
+                    create_failed.append('    {0}'.format(fmt_info))
+        return create_completed, create_failed
 
     def get_results_send(self):
-        try:
-            self._send_completed = []
-            self._send_failed = []
-            for book in self._books:
-                if book.status is book.FAIL:
-                    self._send_failed.append('{0}: {1}'.format(book.title_and_author, book.status_message))
-                    continue
-                fmts_completed = []
-                fmts_failed = []
-                if self._create_send_xray and hasattr(book, '_xray_send_status'):
-                    if book.xray_send_status == book.FAIL:
-                        if hasattr(book, '_xray_send_fmt'):
-                            fmts_failed.append('X-Ray ({0}): {1}'.format(book.xray_send_fmt, book.xray_send_status_message))
-                        else:
-                            fmts_failed.append('X-Ray: {0}'.format(book.xray_send_status_message))
+        '''Gets send results'''
+        send_completed = []
+        send_failed = []
+        for book in self._books:
+            if book.status is FAIL:
```

Comment:
(this really relates to an easrlier question on why we made this change)  To be this is more fragile.  Say you have two modules that both define some FAIL value, but do it differently.  Now we are just referring to 'FAIL' without it being obvious that it is the 'FAIL' that relates to books, so if somewhere else you naively do something like 'global.status is FAIL' it looks fine at a glance but is wrong.  Having 'book.FAIL' and 'global.FAIL' would make things clearer.

[this isn't a current problem, we only have FAIL definted once, just giving an example of how it could go wrong in future with this sort of naming]